src/app/core/mods/zaapSearchFilter/zaapSearchFilter.ts

```diff
@@ -0,0 +1,116 @@
+import { Logger } from "app/core/electron/logger.helper";
+
+import { Mod } from "../mod";
+
+
+export class ZaapSearchFilter extends Mod {
+
+    private styleTag: HTMLDivElement;
+    private zaapSearchContainer: HTMLDivElement;
+    private zaapSearchInput : HTMLInputElement;
+    private inputPlaceholder: string;
+    
+
+    startMod(): void {
+        this.params = this.settings.option.vip.general.zaapsearchfilter;
+        this.inputPlaceholder = this.translate.instant('app.option.vip.zaapsearchfilter.placeholder');
+
+        if(this.params){
+            Logger.info('- enable ZaapSearchFilter');
+
+            this.on(this.wGame.dofus.connectionManager, 'ZaapListMessage', (msg: any) => {
+                this.createSearchFilter();
+            });
+    
+            this.on(this.wGame.connectionManager, 'LeaveDialogMessage', (msg: any) => {
+                this.resetSearchFilter();
+            });
+        }
+    }
+
+
+    private createSearchFilter(): void{
+        this.injectInputInDom();
+
+        this.zaapSearchInput.addEventListener("change", () => {
+            let zaapWanted = this.zaapSearchInput.value;
+            let zaapList = this.wGame.document.getElementsByClassName('zaapBodyHeight__custom')[0]
+                            .getElementsByClassName('row');
+
+            for(let i = 0; i < zaapList.length; i++){
+                let currentZaap = zaapList[i];
+                let destination = currentZaap.getElementsByClassName('destinationName')
+				
+				if(destination.length > 0){
+
+                    currentZaap.style.display = "none";
+
+					if( currentZaap.innerHTML.includes( zaapWanted ) ){
+                        currentZaap.style.display = "block";
+                    } 
+                }
+            }
+        });
+    }
+
+
+    private injectInputInDom(): void{
+        this.styleTag = this.wGame.document.createElement('style');
+        this.wGame.document.getElementsByTagName('head')[0].appendChild(this.styleTag);
+ 
+        this.styleTag.innerHTML += `
```

Comment:
**Suggestion & Readability**

Consider moving this content away in another file, and import the variable in this file. This way, you save 20 lines of space in this file